change/@ni-nimble-components-15e8f878-a870-4090-8bf6-7c45c65195ba.json

```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "Add virtualization to table",
```

Comment:
Here's the testing I've done by playing with the tabledemo page on macOS:

Good news: 
Firefox - scrolling feels snappy, no flickering. Performance and memory profiling show we're nowhere near 100% cpu and consistently under 20MB of memory usage.

Bad news: 
Safari - table renders with no data. Console shows this error repeated 60 times (once per cell?):
```
[Error] TypeError: undefined is not an object (evaluating 'n.appendChild')
	runGuarded (polyfills.54ab400a48eba0a6.js:1:2165)
	appendChild
	appendTo (main.a4d212b3f0e17485.js:1:264961)
	render (main.a4d212b3f0e17485.js:1:268254)
	renderTemplate (main.a4d212b3f0e17485.js:1:244195)
	finishInitialization (main.a4d212b3f0e17485.js:1:243793)
	onConnectedCallback (main.a4d212b3f0e17485.js:1:242794)
	connectedCallback (main.a4d212b3f0e17485.js:1:244616)
	connectedCallback (main.a4d212b3f0e17485.js:1:257647)
	runGuarded (polyfills.54ab400a48eba0a6.js:1:2092)
	insertBefore
	insertBefore (main.a4d212b3f0e17485.js:1:265051)
	d$ (main.a4d212b3f0e17485.js:1:261462)
	handleChange (main.a4d212b3f0e17485.js:1:263178)
	notify (main.a4d212b3f0e17485.js:1:231601)
	call (main.a4d212b3f0e17485.js:1:233925)
	i (main.a4d212b3f0e17485.js:1:229445)
	o (main.a4d212b3f0e17485.js:1:229526)
	(anonymous function) (polyfills.54ab400a48eba0a6.js:1:28207)
	runTask (polyfills.54ab400a48eba0a6.js:1:2590)
	invokeTask (polyfills.54ab400a48eba0a6.js:1:8243)
```

Looks like this error is actually pre-existing (reproduces in the demo page build from main) so we don't necessarily need to fix it for this PR, but it would be nice to do perf testing once it's fixed. I'm happy to jump on a call to debug if you want or I can file a bug.

![image](https://user-images.githubusercontent.com/10500124/213289316-67cd9133-8c24-4f30-a444-6299bdb0dad1.png)

Reply:
More info on the Safari issue (will update if I learn more):
In Firefox, `cellTemplateChanged` is called many times but `this.isConnected` is false.
In Safari, `cellTemplateChanged` is being called many times but `this.isConnected` is true. However `this.cellContentContainer` is `undefined`. This results in a call within FAST code to `view.appendTo(undefined)` and thus `undefined.appendChild()`.

Reply:
Since the issue is pre-existing I'd vote for creating a separate bug/work item instead of addressing it in this PR.

I tried it out in the latest Playwright-provided Webkit browser app on Windows, and it reproduces there too, so hopefully that makes it easier for us non-Mac-users to debug it.
(That build is here and it supports Inspect/ DevTools: https://playwright.azureedge.net/builds/webkit/1751/webkit-win64.zip )

Reply:
#984